pandas/_libs/tslibs/conversion.pyx

```diff
@@ -297,9 +297,18 @@ cdef _TSObject convert_to_tsobject(object ts, tzinfo tz, str unit,
         if ts == NPY_NAT:
             obj.value = NPY_NAT
         else:
-            ts = cast_from_unit(ts, unit)
-            obj.value = ts
-            pandas_datetime_to_datetimestruct(ts, NPY_FR_ns, &obj.dts)
+            in_reso = abbrev_to_npy_unit(unit or "ns")
```

Comment:
could just make ns the default?  either way, can you make the check explicitly for None so that "" doesn't accidentally get in here